Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -996,6 +399,200 @@ bool CGImageIsMask(CGImageRef image) {
  @Notes
 */
 CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* components) {
+    RETURN_NULL_IF(!image);
     UNIMPLEMENTED();
     return StubReturn();
-}
\ No newline at end of file
+}
+
+#pragma region WIC_HELPERS
+
+CGImageRef _CGImageGetImageFromData(void* data, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_NULL, data, length);
+}
+
+CGImageRef _CGImageLoadGIF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatGif, bytes, length);
+}
+
+CGImageRef _CGImageLoadBMP(void* bytes, size_t length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatBmp, bytes, length);
+}
+
+CGImageRef _CGImageLoadTIFF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatTiff, bytes, length);
+}
+
+CGImageRef _CGImageLoadPNG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatPng, bytes, length);
+}
+
+CGImageRef _CGImageLoadJPEG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatJpeg, bytes, length);
+}
+
+CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void* bytes, int length) {
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+
+    ComPtr<IWICBitmapDecoder> pDecoder;
+
+    if (decoderCls != GUID_NULL) {
+        ComPtr<IStream> spStream;
+        RETURN_NULL_IF_FAILED(::CreateStreamOnHGlobal(NULL, FALSE, &spStream));
```

Comment:
>CreateStreamOnHGlobal [](start = 32, length = 21)

seems ... weird to create stream on a global handle with NULL. Is this really the preferred mechanism to make an empty stream?